genshin/models/genshin/chronicle/activities.py

```diff
@@ -228,6 +292,13 @@ class Activities(APIModel):
     martial_legend: typing.Optional[Activity[typing.Any]] = None
     chess: typing.Optional[Activity[typing.Any]] = None
 
+    @pydantic.validator("finish_time", pre=True)
```

Comment:
validators go on the models with the relevant field.

Reply:
you mean put it after the model?

Reply:
Put it into story record and challenge record models.